scripts/o3de/o3de/engine_template.py

```diff
@@ -1633,16 +1647,19 @@ def create_project(project_path: pathlib.Path,
         template_restricted_platform_relative_path = ''
 
     # if no project path, error
-    if not project_path:
-        logger.error('Project path cannot be empty.')
+    if not project_path and not project_name:
+        logger.error('Project path and Project name cannot both be empty.')
         return 1
-    project_path = project_path.resolve()
-    if not os.path.isabs(project_path):
+    if not project_path and project_name:
         default_projects_folder = manifest.get_registered(default_folder='projects')
-        new_project_path = default_projects_folder / project_path
-        logger.info(f'Project Path {project_path} is not a full path, we must assume its relative'
+        new_project_path = default_projects_folder / project_name
+        logger.info(f'Only Project Name {project_name} specified, we must assume its relative'
                     f' to default projects path = {new_project_path}')
         project_path = new_project_path
+        # project name is now the last component of the project_path, so if someone enters some/path/toproject
+        # the destination is <default folder>/some/path/toproject which would make the name of the gem "toproject"
+        project_name = os.path.basename(project_path)
```

Comment:
The project_name shouldn't be set here as the user supplied via the command line option.
This code is within the if block when `project_name` is valid and therefore we shouldn't override the value.